addons/knowledge/static/src/js/widgets/knowledge_dialogs.js

```diff
@@ -0,0 +1,186 @@
+odoo.define('knowledge.dialogs', function (require) {
+'use strict';
+
+const Dialog = require('web.Dialog');
+const { _t } = require('web.core');
+
+const MoveArticleToDialog = Dialog.extend({
+    template: 'knowledge.knowledge_move_article_to_modal',
+    /**
+     * @override
+     * @param {Widget} parent
+     * @param {Object} options
+     * @param {Object} props
+     */
+    init: function (parent, options, props) {
+        // Set default options:
+        options.title = options.title || _t('Move an Article');
+        options.buttons = options.buttons || [{
+            text: _t('Ok'),
+            classes: 'btn-primary',
+            click: this.save.bind(this)
+        }, {
+            text: _t('Cancel'),
+            close: true
+        }];
+        this._super(...arguments);
+        this.props = props;
+        // Set template variables:
+        const { state } = props;
```

Comment:
Do we really need the whole article state?
This seems like a very heavy object to receive and it's hard to see here what it would contain.

Can't we instead require a few key variables within the options? articleName, articleId, ... ?

(Side-note: use camelCase everywhere possible is JS)